src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -678,7 +679,7 @@ private static String EscapeIDName(String inID) throws SQLServerException {
              * when user provides a different catalog than the one they're currently connected to. Will throw exception
              * when it's different and do nothing if it's the same/null.
              */
-            synchronized (SQLServerDatabaseMetaData.class) {
+            synchronized (this) {
```

Comment:
I think this synchronized block should be on a class level, to ensure that no race condition occurs when trying to initialize getColumnsDWColumns